frontend/src/views/CRAllotment.vue

```diff
@@ -0,0 +1,194 @@
+          <v-btn flat v-for="language in languages" :key="language" :value="language">{{language}}</v-btn>
+        </v-btn-toggle>
+      </v-layout>
+      <!-- List -->
+      <v-layout row class="py-2">
+        <v-btn-toggle v-model="filter.list" v-if="lists">
+          <v-btn flat v-for="list in lists" :key="list" :value="list">{{list}}</v-btn>
+        </v-btn-toggle>
+        <p v-else>Loading lists…</p>
+      </v-layout>
+      <!-- Files -->
+      <template v-if="filter.list && filter.languages.length">
+        <template v-if="files">
+          <template v-if="files.length > 0">
+            <template v-for="(file, index) in files">
+              <div :key="file.filename">
+                <v-divider v-if="index > 0 && files[index - 1].date !== file.date"/>
+                <v-layout align-center>
+                  <v-checkbox
+                    :style="{ flex: 'none' }"
+                    v-model="allotment.files"
+                    :value="file"
+                    :loading="!files"
+                    class="mr-2"
+                  >
+                    <code slot="label">{{ file.filename }}</code>
+                  </v-checkbox>
+                  <span>{{ file.date || "No date" }} {{ file.language || "No language" }} {{ file.notes }}</span>
+                </v-layout>
+              </div>
+            </template>
+          </template>
+          <p v-else>No spare files found for selected language in {{filter.list}} list.</p>
+        </template>
+        <p v-else>Loading files…</p>
+      </template>
+      <p v-else>Choose list and language to select files.</p>
+      <!-- Comment -->
+      <v-textarea v-model="allotment.comment" box label="Comment" rows="3"></v-textarea>
+      <!-- Buttons -->
+      <v-btn @click="allot" :loading="submissionStatus === 'inProgress'">Allot</v-btn>
+    </v-form>
+    <v-alert
+      v-else
+      :value="submissionStatus === 'complete'"
+      type="success"
+      transition="scale-transition"
+    >
+      <h4 class="alert-heading">Allotted successfully</h4>
+      <p class="mb-0">
+        <v-btn @click="reset">Make another allotment</v-btn>
+      </p>
+    </v-alert>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Watch } from "vue-property-decorator";
+import firebase from "firebase/app";
+import "firebase/functions";
+import _ from "lodash";
+
+import { initialAllotment } from "@/utility";
+
+@Component({
+  name: "CRAllotment"
+})
+export default class CRAllotment extends Vue {
+  assignees: any = null;
+  languages: string[] = ["English", "Hindi", "Bengali", "None"];
+  lists = null;
+  files: any = null;
+  filter = CRAllotment.initialFilter();
+  allotment: any = initialAllotment();
+  submissionStatus: string | null = null;
+
+  static initialFilter() {
+    return {
+      languages: [] as string[],
+      list: null
+    };
+  }
+
+  async mounted() {
+    this.getAssignees();
+    this.getLists();
```

Comment:
Will these two methods work in parallel? I don't think so because of `await` is used inside them. The result is that lists are loaded only after assignees are loaded. That is why I was using `then` in `SQRAllotment`. Correct me if I'm wrong. If I'm correct then let's bring the code back to this method. I'm not against code splitting, but in this case this method is not that huge, it's easily comprehensible without splitting.

Reply:
It appears getLists also fetches data without waiting for assignees.

Reply:
ok then.